src/components/fx/hover.js

```diff
@@ -1993,19 +1993,33 @@ function getCoord(axLetter, winningPoint, fullLayout) {
     var ax = winningPoint[axLetter + 'a'];
     var val = winningPoint[axLetter + 'Val'];
 
+    var trace = winningPoint.trace;
+    var cd = winningPoint.cd;
+    var d = cd[winningPoint.index];
+
     if(ax.type === 'category') val = ax._categoriesMap[val];
     else if(ax.type === 'date') {
-        var period = winningPoint[axLetter + 'Period'];
-        val = ax.d2c(period !== undefined ? period : val);
+        var periodalignment = trace[axLetter + 'periodalignment'];
+        if(periodalignment) {
+            var start = d[axLetter + 'Start'];
+            var end = d[axLetter + 'End'];
+            var diff = end - start;
+            if(periodalignment === 'end') {
+                val += diff;
+            } else if(periodalignment === 'middle') {
+                val += diff / 2;
+            }
+        }
+
+        val = ax.d2c(val);
     }
 
-    var cd0 = winningPoint.cd[winningPoint.index];
-    if(cd0 && cd0.t && cd0.t.posLetter === ax._id) {
+    if(d && d.t && d.t.posLetter === ax._id) {
```

Comment:
This code has been around for ages but I'm just noticing it now and it confused the heck out of me: normally the `t` object only exists on the first entry in `cd` (that's why the var we extracted it from was called `cd0`) because it contains extra calculated info that applies to the whole trace. And at the `calc` step that's [true of box and violin too](https://github.com/plotly/plotly.js/blob/51eded41ac7433a9193fe0c21439dbc5f92b6dd6/src/traces/box/calc.js#L278-L298). But then during the `plot` step it appears these traces [add `t` to all `cd` entries](https://github.com/plotly/plotly.js/blob/51eded41ac7433a9193fe0c21439dbc5f92b6dd6/src/traces/box/plot.js#L172-L180) so that if you draw points, the one box the points are for will look like a whole trace to the reused scatter code with which we're drawing the points.

So yes, this works... but only due to a quirk of our drawing code and only for box and violin, so it feels fragile. Nonblocking, but when we need `t` we should really get it from `cd[0]`.

Reply:
What you suggesting here is renaming `d` back to `cd0`? Or I should simply add a comment above `cd0.d`?

Reply:
I'm saying instead of assuming box & violin will have `t` in every element of `cd` (`d` is `cd[winningPoint.index]` and `winningPoint.index` need not be zero) we should be using `cd[0].t`.

Reply:
Addressed in https://github.com/plotly/plotly.js/pull/5864/commits/3e52b941be9b9e384a9c345bc2ab6a16076926fe